src/main/java/graphql/execution/ValuesResolver.java

```diff
@@ -388,30 +388,38 @@ private Object externalValueToLiteralForObject(GraphqlFieldVisibility fieldVisib
      */
     private Map<String, Object> externalValueToInternalValueForVariables(GraphQLSchema schema,
                                                                          List<VariableDefinition> variableDefinitions,
-                                                                         Map<String, Object> rawVariables
-    ) {
+                                                                         Map<String, Object> rawVariables) {
         GraphqlFieldVisibility fieldVisibility = schema.getCodeRegistry().getFieldVisibility();
         Map<String, Object> coercedValues = new LinkedHashMap<>();
         for (VariableDefinition variableDefinition : variableDefinitions) {
-            String variableName = variableDefinition.getName();
-            GraphQLType variableType = TypeFromAST.getTypeFromAST(schema, variableDefinition.getType());
-            assertTrue(variableType instanceof GraphQLInputType);
-            // can be NullValue
-            Value defaultValue = variableDefinition.getDefaultValue();
-            boolean hasValue = rawVariables.containsKey(variableName);
-            Object value = rawVariables.get(variableName);
-            if (!hasValue && defaultValue != null) {
-                Object coercedDefaultValue = literalToInternalValue(fieldVisibility, variableType, defaultValue, Collections.emptyMap());
-                coercedValues.put(variableName, coercedDefaultValue);
-            } else if (isNonNull(variableType) && (!hasValue || value == null)) {
-                throw new NonNullableValueCoercedAsNullException(variableDefinition, variableType);
-            } else if (hasValue) {
-                if (value == null) {
-                    coercedValues.put(variableName, null);
-                } else {
-                    Object coercedValue = externalValueToInternalValue(fieldVisibility, variableType, value);
-                    coercedValues.put(variableName, coercedValue);
+            try {
```

Comment:
The fundamental change is to move the `CoercingParseValueException` catch one function earlier in the chain, to `externalValueToInternalValueForVariables`.

Although it appears many lines are changing, it's mostly only indentation for the try catch block.

Reply:
The rationale for moving the catch: the source location is available here, whereas it is not passed down to the next function `externalValueToInternalValue`.